src/test/compile-fail/rfc-1937-termination-trait/termination-trait-main-i32.rs

```diff
@@ -8,6 +8,7 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-fn main() -> i32 { //~ ERROR main function has wrong type [E0580]
+fn main() -> i32 {
+//~^ ERROR the trait bound `i32: std::process::Termination` is not satisfied [E0277]
```

Comment:
Hmm, this is not your fault, but this error message seems less clear than before.

@estebank -- can we improve this readily enough with a `#[rustc_on_unimplemented]` annotation perhaps?

Reply:
Yes, you could annotate `std::process::Termination` using `#[rustc_on_unimplemented]` to replace the error message to something appropriate. The only limitation is not knowing wether the requirement came from a return type or anywhere else, so the message will have to be more generic than it is now.

Reply:
It looks like [there is already a `#[rustc_on_unimplemented]` annotation](https://github.com/rust-lang/rust/blob/e5a55e74405dedf8bc0744300a8c506eea94bc18/src/libstd/process.rs#L1445), but it is not used in the error message for some reason.

Reply:
Nevermind, once I cleaned my rustc build the note showed up. But since it's a label, not the primary error message, I had to use the `error-message:` directive to match against it.

Reply:
For what it's worth, you have further control on the `rustc_on_unimplemented` output by [being able to set a separate message and label](https://github.com/rust-lang/rust/pull/47613/files#diff-6f956da589a7eec6332ba1540c629560R533), as well as customize either one for specific types.

Reply:
> since it's a label, not the primary error message, I had to use the error-message: directive to match against it.

You didn't need to use `error-message`, you could use 
```
//~^ NOTE `main` can only return types that implement std::process::Termination, not `i32`
```
instead for labels.